go-apps/meep-metrics-engine/api/v2/swagger.yaml

```diff
@@ -272,112 +315,268 @@ definitions:
+        eventQueryParams:
+          scope:
+            duration: "10s"
+            limit: 60
+          fields:
+          - "lat"
+          - "ul"
+          - "dl"
+          - "ulos"
+          - "dlos"
+          tags:
+          - name: "src"
+            value: "ue1-iperf"
+          - name: "dest"
+            value: "zone1-fog1-iperf"
+        callbackReference: "{\"notifyURL\":\"http://clientApp.example.com/metric_network_notifications/123456\"\
+          }"
+        clientCorrelator: "83"
+        subscriptionId: "1.2556487E7"
+      - resourceURL: "http://localhost:8291/v2/subscriptions/network/subscription123"
+        eventQueryParams:
+          scope:
+            duration: "10s"
+            limit: 60
+          fields:
+          - "lat"
+          - "ul"
+          - "dl"
+          - "ulos"
+          - "dlos"
+          tags:
+          - name: "src"
+            value: "ue1-iperf"
+          - name: "dest"
+            value: "zone1-fog1-iperf"
+        callbackReference: "{\"notifyURL\":\"http://clientApp.example.com/metric_network_notifications/123456\"\
+          }"
+        clientCorrelator: "83"
+        subscriptionId: "1.2556487E7"
+  NetworkSubscription:
     type: "object"
     properties:
-      name:
+      subscriptionId:
         type: "string"
-        example: "network metrics"
-        description: "Response name"
-      columns:
-        type: "array"
-        example: "time,lat,tput,loss"
-        description: "columns included in response based on queried values"
-        items:
-          type: "string"
-          example: "lat,tput,loss"
-          description: "Queried value"
-      values:
-        type: "array"
-        items:
-          $ref: "#/definitions/NetworkMetrics"
-    description: "Network metrics query response"
+        example: "1.2556487E7"
+        description: "Subscription identifier"
+      clientCorrelator:
+        type: "string"
+        example: "83"
+        description: "Uniquely identifies this create subscription request. If there\
+          \ is a communication failure during the request, using the same clientCorrelator\
+          \ when retrying the request allows the operator to avoid creating a duplicate\
+          \ subscription."
+      callbackReference:
+        $ref: "#/definitions/NetworkCallbackReference"
+      resourceURL:
+        type: "string"
+        format: "uri"
+        example: "http://localhost:8291/v2/subscriptions/network/subscription123"
+        description: "Self referring URL."
+      networkQueryParams:
+        $ref: "#/definitions/NetworkQueryParams"
+      period:
+        type: "integer"
+        example: 1
+        description: "Notification interval in seconds, disabled if set to 0"
+      subscriptionType:
+        type: "string"
+        example: "period"
+        description: "Type of subscription triggering notifications"
+        enum:
+        - "period"
```

Comment:
To avoid having to use 'event-based', we decided to go with a noun instead of an adjective here. Therefore currently supported type is only 'period', but should eventually be 'period' or 'event'.